packages/cli/src/commands/validator/status.ts

```diff
@@ -0,0 +1,117 @@
+import { Address } from '@celo/contractkit'
+import { flags } from '@oclif/command'
+import { cli } from 'cli-ux'
+import { range } from 'lodash'
+import * as rlp from 'rlp'
+import { Block } from 'web3/eth/types'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { Flags } from '../../utils/command'
+
+export default class ValidatorOnline extends BaseCommand {
+  static description =
+    'Show information about whether the validator signer is elected and validating. This command will check that the validator meets the registration requirements, and its signer is currently elected and actively signing blocks.'
+
+  static flags = {
+    ...BaseCommand.flags,
+    signer: Flags.address({
+      description: 'address of the validator to check if elected and validating',
+    }),
+    validator: Flags.address({
+      description: 'address of the signer to check if elected and validating',
+    }),
+    lookback: flags.integer({
+      description: 'how many blocks to look back for signer activity',
+      default: 50,
+    }),
+  }
+
+  static examples = [
+    'status --validator 0x5409ED021D9299bf6814279A6A1411A7e866A631',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c --lookback 100',
+  ]
+
+  requireSynced = true
+
+  async run() {
+    const { flags } = this.parse(ValidatorOnline)
+
+    if (!flags.validator && !flags.signer) {
+      this.error('Either validator or signer must be specified')
+    }
+
+    // Determine the validator signer to check.
+    let signer: Address
+    if (flags.validator) {
+      const accounts = await this.kit.contracts.getAccounts()
+      signer = await accounts.getValidatorSigner(flags.validator)
+      console.info(`Identified ${signer} as the authorized validator signer`)
+
+      // If the user specified both, verify that it matches what is stored on chain.
+      if (flags.signer && signer.toLowerCase() !== flags.signer.toLowerCase()) {
+        this.error(`${flags.signer} is not the authorized signer for ${flags.validator}`)
+      }
+    } else {
+      signer = flags.signer!
+    }
+
+    // Use redundant checks to help the user diagnose issues.
+    await newCheckBuilder(this, signer)
+      .canSignValidatorTxs()
+      .signerMeetsValidatorBalanceRequirements()
+      .signerAccountIsValidator()
+      .runChecks()
+
+    // Determine if the signer is elected, and get their index in the validator set.
+    const election = await this.kit.contracts.getElection()
+    const signers = await election.getCurrentValidatorSigners()
+    const signerIndex = signers.map((a) => a.toLowerCase()).indexOf(signer.toLowerCase())
+    if (signerIndex < 0) {
+      // Determine whether the signer will be elected at the next epoch to provide a helpful error.
```

Comment:
We could add a command like `election:validator-is-elected`, but I personally don't see a ton of value in that because `election:current` and `election:run` exist and combined with grep you can tell whether a specific validator is in the set.